addons/ai/fnc_searchNearby.sqf

```diff
@@ -9,51 +9,56 @@ Parameters:
 
 Example:
     (begin example)
-    [group player] spawn CBA_fnc_searchNearby
+    [group player] call CBA_fnc_searchNearby
     (end)
 
 Returns:
     Nil
 
 Author:
-    Rommel
+    Rommel, SilentSpike
 
 ---------------------------------------------------------------------------- */
 
 params ["_group"];
-_group = _group call CBA_fnc_getgroup;
-_group lockwp true;
-private ["_leader","_behaviour"];
-_leader = leader _group;
-_behaviour = behaviour _leader;
-_group setbehaviour "combat";
-
-(_leader call CBA_fnc_getnearestbuilding) params ["_building", "_indices"];
-_group setformdir ([_leader, _building] call bis_fnc_dirto);
-
-if (_leader distance _building > 500) exitwith {_group lockwp false};
-
-private ["_count","_units"];
-_units = units _group;
-_count = (count _units) - 1;
-
-while {_indices > 0 && {_count > 0}} do {
-    sleep 10;
-    while {_indices > 0 && {_count > 0}} do {
-        private "_unit";
-        _unit = _units select _count;
-        if (unitready _unit) then {
-            _unit commandmove (_building buildingpos _indices);
-            _indices = _indices - 1;
-        };
-        _count = _count - 1;
+_group = _group call CBA_fnc_getGroup;
+_group lockWP true;
+
+private _building = nearestBuilding (leader _group);
+if ((leader _group) distanceSqr _building > 250e3) exitwith {_group lockWP false};
+
+[_group,_building] spawn {
+    params ["_group","_building"];
+    private _behaviour = behaviour (leader _group);
+
+    // Prepare group to search
+    _group setBehaviour "Combat";
+    _group setFormDir ((leader _group) getDir _building);
+
+    // Search while there are still available positions
+    private _positions = _building buildingPos -1;
+    while {!(_positions isEqualTo [])} do {
+        // Abort search if the group has no units left
+        if ((units _group) isEqualTo []) exitWith {};
+
+        // Send all available units to the next available position

```

Comment:
Okay, got the waypoint solution working cleanly - it's actually preferable to using AI commands because it also takes care of spare group members too and allows me to put the post-search code into the completion statement instead of running a loop :smile: